build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -0,0 +1,2104 @@
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_MANIFEST_SAVE_THRESHOLD =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_DEFAULT => 1073741824,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_RESUME =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => true,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_START_FAST =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_STOP_AUTO =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    # Expire options
+    #-------------------------------------------------------------------------------------------------------------------------------
+    &CFGBLDOPT_RETENTION_ARCHIVE =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_REQUIRED => false,
+        &CFGBLDDEF_RULE_ALLOW_RANGE => [CFGBLDDEF_DEFAULT_RETENTION_MIN, CFGBLDDEF_DEFAULT_RETENTION_MAX],
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_EXPIRE => {},
+        }
+    },
+
+    &CFGBLDOPT_RETENTION_ARCHIVE_TYPE =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_STRING,
+        &CFGBLDDEF_RULE_DEFAULT => CFGBLDOPTVAL_BACKUP_TYPE_FULL,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_EXPIRE => {},
+        },
+        &CFGBLDDEF_RULE_ALLOW_LIST =>
+        {
+            &CFGBLDOPTVAL_BACKUP_TYPE_FULL => 1,
+            &CFGBLDOPTVAL_BACKUP_TYPE_DIFF => 1,
+            &CFGBLDOPTVAL_BACKUP_TYPE_INCR => 1
+        }
+    },
+
+    &CFGBLDOPT_RETENTION_DIFF =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_REQUIRED => false,
+        &CFGBLDDEF_RULE_ALLOW_RANGE => [CFGBLDDEF_DEFAULT_RETENTION_MIN, CFGBLDDEF_DEFAULT_RETENTION_MAX],
+        
```

Comment:
These are different because they were added recently after I created the defaults (for the s3 feature).  I haven't gone back and made wholesale changes to the other options, and I definitely won't in this commit.  If we want to make changes to this logic I think that should be in another commit as well.